packages/cli/src/commonlib/common/cicdPlatformDetector.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import { CliConfigRunFrom } from "../../userSetttings";
+
+export function tryDetectCICDPlatform(): CliConfigRunFrom {
+  if (process.env.GITHUB_ACTIONS === "true") {
+    // https://docs.github.com/cn/actions/learn-github-actions/environment-variables#default-environment-variables
+    return CliConfigRunFrom.GitHub;
+  } else if (process.env.BUILD_SOURCEBRANCHNAME && process.env.AGENT_BUILDDIRECTORY) {
```

Comment:
If not pre-defined env for azdo, how about moving this condition after jenkins? `JENKINS_URL` sounds more platform-specific.

Reply:
Good suggestions, Will do.